src/System.Net.Http/src/System/Net/Http/DiagnosticsHandler.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http
+{
+    /// <summary>
+    /// DiagnosticHandler notifies DiagnosticSource subscribers about outgoing Http requests
+    /// </summary>
+    internal sealed class DiagnosticsHandler : DelegatingHandler
+    {
+        /// <summary>
+        /// DiagnosticHandler constructor
+        /// </summary>
+        /// <param name="innerHandler">Inner handler: Windows or Unix implementation of HttpMessageHandler. 
+        /// Note that DiagnosticHandler is the latest in the pipeline </param>
+        public DiagnosticsHandler(HttpMessageHandler innerHandler) : base(innerHandler)
+        {
+        }
+
+        internal static bool IsEnabled()
+        {
+            return s_diagnosticListener.IsEnabled(DiagnosticsHandlerLoggingStrings.ResponseWriteName);
```

Comment:
We should be clear that the number of people using DiagnosticSource is limited (probably less than 100 worldwide).   Moreover it is used in the most perf-critical parts of the system.   I think it is reasonable to bias things toward perf at the expense of usability, especially if that usability can be fixed with some reasonable docs.   In the not too-distant furture (months) we shodul have having the and source server set up so that people can frankly just debug into things if they run into a problem with a very good experience.   That is probably BETTER than docs because it will always be accurate, and clear.    

There is a pretty good chance that we will use this trick other places (basically form 'groups' of events where one is considered the 'master' of the others.   

@lmolkova  can you add to the description of the 'IsEnabled' callback the fact that some events (typically a stop event) might used as the 'gate' for other events (e.g. the cooresponding start event).